pycheribuild/projects/cross/ros2.py

```diff
@@ -0,0 +1,99 @@
+#
+# Copyright (c) 2020 Michael Dodson
+# All rights reserved.
+#
+# This software was developed by SRI International and the University of
+# Cambridge Computer Laboratory under DARPA/AFRL contract FA8750-10-C-0237
+# ("CTSRD"), as part of the DARPA CRASH research programme.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+# OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+# SUCH DAMAGE.
+#
+
+from os.path import isdir
+from .crosscompileproject import *
+# from ..project import ReuseOtherProjectRepository
+
+
+class BuildRos2(CrossCompileCMakeProject):
+    project_name = "ros2"
+    repository = GitRepository("https://github.com/dodsonmg/ros2_dashing_minimal.git", default_branch="master", force_branch=True)
+
+    # atm, we build and install in the sourceDir.
+    # it may eventually be useful to install to rootfs or sysroot depending on whether we want to use ROS2
+    # as a library for building other applications using cheribuild
+    # therefore, these _install_dir don't do anything, but cheribuild requires them
+    native_install_dir = DefaultInstallDir.IN_BUILD_DIRECTORY
+    cross_install_dir = DefaultInstallDir.ROOTFS
+
+    dependencies = ["poco"]
+
+    def _ignore_packages(self):
+        packages = ["src/ros2/rcl_logging/rcl_logging_log4cxx"] # relative to self.sourceDir
+        for package in packages:
+            cmdline = ["touch"]
+            cmdline.append(str(self.sourceDir / package / "COLCON_IGNORE"))
+            self.run_cmd(cmdline, cwd=self.sourceDir)
+        return
+
+    def _make_src_dir(self):
```

Comment:
There is a self.makefiles() in the parent class already

Reply:
used `self.makedirs`